Solves a bug when typing Control+LeftArrow and other similar combinations

loudenvier opened the pull request:
When typing Control+LeftArrow (and other similar key combinations) the \0 char was being written to the console which was causing havoc in typing. This commit fixes it by preventing writing the \0 char, but, maybe we should prevent writing non-printable characters altogether

Latency commented:
Check on my fork for a fix.  I can merge it there since owner is inactive on this repo 

I will be back from PTO after Oct 2.  I can merge and publish it after that date at earliest.

loudenvier commented:
> Check on my fork for a fix. I can merge it there since owner is inactive on this repo
> 
> I will be back from PTO after Oct 2. I can merge and publish it after that date at earliest.

I'm just submitting this PR because way too many people uses ReadLine and it's a trivial merge for the owner. I've moved on with lots of changes now on my own fork and added more commands to make it more GNU ReadLine compliant. I've published a package (still behind of the latest updates) under Loudenvier.ReadLine (https://www.nuget.org/packages/Loudenvier.ReadLine)

A curiosity: I see that you changed the code to use Allman indentation and I hate Allman formatting with a passion!!! :-) I don't like to loose vertical space in code, and with all the syntax highlighting we have today, a more concise code format is not less readable than Allman's (I think it's more readable as you can see more code on the screen without scrolling). I use Allman only for class and namespace declarations (in fact, I've moved on to file-scoped namespaces so not even them uses Allman anymore). But what I DO love in all of this is how people enjoy different indentation styles and there is not ONE style to rule them all!

Happy coding!

Latency commented:
Hello,

I will investigate the formatting and review the PR.

Thx,
Latency




________________________________
From: Felipe Machado ***@***.***>
Sent: Thursday, September 26, 2024 6:51:04 AM
To: tonerdo/readline ***@***.***>
Cc: Latency McLaughlin ***@***.***>; Comment ***@***.***>
Subject: Re: [tonerdo/readline] Solves a bug when typing Control+LeftArrow and other similar combinations (PR #70)


Check on my fork for a fix. I can merge it there since owner is inactive on this repo

I will be back from PTO after Oct 2. I can merge and publish it after that date at earliest.

I'm just submitting this PR because way too many people uses ReadLine and it's a trivial merge for the owner. I've moved on with lots of changes now on my own fork and added more commands to make it more GNU ReadLine compliant. I've published a package (still behind of the latest updates) under Loudenvier.ReadLine (https://www.nuget.org/packages/Loudenvier.ReadLine)

A curiosity: I see that you changed the code to use Allman indentation and I hate Allman formatting with a passion!!! :-) I don't like to loose vertical space in code, and with all the syntax highlighting we have today, a more concise code format is not less readable than Allman's (I think it's more readable as you can see more code on the screen without scrolling). I use Allman only for class and namespace declarations (in fact, I've moved on to file-scoped namespaces so not even them uses Allman anymore). But what I DO love in all of this is how people enjoy different indentation styles and there is not ONE style to rule them all!

Happy coding!

—
Reply to this email directly, view it on GitHub<https://github.com/tonerdo/readline/pull/70#issuecomment-2376605524>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAGAOSFBK64RYVXAHPGRO43ZYPRJPAVCNFSM6AAAAABO3KFCMOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGNZWGYYDKNJSGQ>.
You are receiving this because you commented.Message ID: ***@***.***>
